### PR TITLE
Add approximate_pi.cpp file

### PR DESCRIPTION
**Pull request by Qannaf**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1860"></a>

### CODE REVIEW

**Comment by Qannaf** [comment]
This function  approximate pi using the given points

**Review by Qannaf — COMMENTED** [review]
Review

File: math/approximate_pi.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/**
+ * @file
+ * @brief Implementation to  calculate an estimate of the number π (Pi).
```
Comment (Panquesito7) [inline math/approximate_pi.cpp:3]: Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

File: math/approximate_pi.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * We know that the probability of the point being inside the quarter disk is equal to π/4
+ * double approx(vector<Point> &pts) which will use the points pts (drawn at random) to 
+ * return an estimate of the number π
+ * \note This implementation is better than naive recursive or iterative
+ * approach.
+ *
+ * @author [Qannaf AL-SAHMI](https://github.com/Qannaf)
+ */
+
+#include <iostream>  /// for std::cout
```
Comment (Panquesito7) [inline math/approximate_pi.cpp:17]:
```suggestion
#include <iostream>  /// for IO operations
```

File: math/approximate_pi.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+#include <vector>    /// for vector
+#include <stdlib.h>  /// for rand
```
Comment (Panquesito7) [inline math/approximate_pi.cpp:19]:
```suggestion
#include <cstdlib>  /// for rand
```

File: math/approximate_pi.cpp [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ * double approx(vector<Point> &pts) which will use the points pts (drawn at random) to 
+ * return an estimate of the number π
+ * \note This implementation is better than naive recursive or iterative
+ * approach.
+ *
+ * @author [Qannaf AL-SAHMI](https://github.com/Qannaf)
+ */
+
+#include <iostream>  /// for std::cout
+#include <vector>    /// for vector
```
Comment (Panquesito7) [inline math/approximate_pi.cpp:18]:
```suggestion
#include <vector>    /// for std::vector
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: math/approximate_pi.cpp (hunk 1 above)
Comment (Qannaf) [inline math/approximate_pi.cpp:3]: * https://en.wikipedia.org/wiki/File:Pi_30K.gif

Comment (Qannaf) [inline math/approximate_pi.cpp:3]:
```suggestion
 * @brief Implementation to  calculate an estimate of the number π (Pi).
 * https://en.wikipedia.org/wiki/File:Pi_30K.gif
```

**Review by Qannaf — COMMENTED** [review]
commit Review changes

File: math/approximate_pi.cpp [hunk 5]
```diff
@@ -0,0 +1,75 @@
+/**
+ * @file
+ * @brief Implementation to  calculate an estimate of the number π (Pi).
+ * https://en.wikipedia.org/wiki/File:Pi_30K.gif
```
Comment (Panquesito7) [inline math/approximate_pi.cpp:4]:
```suggestion
 * @brief Implementation to calculate an estimate of the [number π (Pi)](https://en.wikipedia.org/wiki/File:Pi_30K.gif).
```

File: math/approximate_pi.cpp [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+ * return an estimate of the number π
+ * \note This implementation is better than naive recursive or iterative
+ * approach.
+ *
+ * @author [Qannaf AL-SAHMI](https://github.com/Qannaf)
+ */
+
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector
+#include <cstdlib>  /// for rand
```
Comment (Panquesito7) [inline math/approximate_pi.cpp:20]:
```suggestion
#include <cstdlib>  /// for std::rand
```

File: math/approximate_pi.cpp [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * @namespace math
+ * @brief Mathematical algorithms
+ */
+namespace math {
+/**
+ * @brief Function to test above algorithm
+ * @param n description
+ * @returns void
+ */
```
Comment (Panquesito7) [inline math/approximate_pi.cpp:31]:
```suggestion
```

File: math/approximate_pi.cpp [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    for (std::size_t i = 0; i < 100000; i++) {
+        math::Point p;
+        p.x = rand() / (double)RAND_MAX; // 0 <= x <= 1
+        p.y = rand() / (double)RAND_MAX; // 0 <= y <= 1
+        rands.push_back(p);
+    }
+    std::cout << math::approximate_pi(rands);          // ~3.14
+    return 0;
+}
+
```
Comment (Panquesito7) [inline math/approximate_pi.cpp:75]:
```suggestion
```

File: math/approximate_pi.cpp [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+            int count =0;    // Points in cercle
+            for(Point p:pts)
+                if(p.x * p.x + p.y*p.y <= 1)
+                    ++count;
+            
+            return 4.0*count/pts.size();
+        }
+    }
+}  // namespace math
+
```
Comment (Panquesito7) [inline math/approximate_pi.cpp:59]: Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or in our [**Discord server**](https://discord.gg/c7MnfGFGa6). 😃

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Almost there! 😃

**Pull request merged**